File: examples/scatter.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char * argv[]) {
    int rank, size, root = 0, i, sendcount;
    float a[16], b[2];

    MPI_Init( &argc, &argv );
    MPI_Comm_rank( MPI_COMM_WORLD, &rank );
    MPI_Comm_size( MPI_COMM_WORLD, &size );

    if (rank == root) {
        for (i = 0; i < 16; i++) {
            a[i] = i;
        }
    }

    sendcount = 2;
    MPI_Scatter ( a, sendcount, MPI_FLOAT,
                  b, sendcount, MPI_FLOAT,
                  root, MPI_COMM_WORLD );

    printf( "%d: b[0]=%f\tb[1]=%f\n",
            rank, b[0], b[1] );

    MPI_Finalize();
    return 0;
}

File: examples/scatter.f90
PROGRAM scatter

USE mpi_f08
IMPLICIT NONE

INTEGER myid, nproc, nsnd, i, root
REAL A(16), B(2)

    CALL MPI_Init()
    CALL MPI_Comm_size(MPI_COMM_WORLD, &
                      nproc)
    CALL MPI_Comm_rank(MPI_COMM_WORLD, &
                      myid)
    root = 0
    IF( myid == root ) THEN
        DO i = 1, 16
            a(i) = REAL(i)
        END DO
    END IF
    nsnd = 2
    CALL MPI_Scatter (a, nsnd, MPI_REAL, b, &
                      nsnd, MPI_REAL, root, &
                      MPI_COMM_WORLD)
    WRITE(*,*) myid, ': b(1)=', b(1), &
                      'b(2)=', b(2)

    CALL MPI_Finalize()

END PROGRAM scatter

